### PR TITLE
Real CUIT Request

### PR DESCRIPTION
**Pull request by VicThorX**
I implement a CUIL validator and real request through `Flurl`

### CODE REVIEW

File: CuitService/Controllers/TaxInfoController.cs.orig [hunk 1]
```diff
@@ -0,0 +1,85 @@
+<<<<<<< HEAD
```
Comment (andresmoschini) [inline CuitService/Controllers/TaxInfoController.cs.orig:1]: Wrong file

File: CuitService/CuitService.csproj.orig [hunk 2]
```diff
@@ -0,0 +1,36 @@
+<Project Sdk="Microsoft.NET.Sdk.Web">
```
Comment (andresmoschini) [inline CuitService/CuitService.csproj.orig:1]: Wrong file

File: CuitService/TaxInfoProviderConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace CuitService
+{
+    public class TaxInfoProviderConfiguration
```
Comment (andresmoschini) [inline CuitService/TaxInfoProviderConfiguration.cs:9]: We have talked about naming it `TaxInfoProviderOptions`

File: CuitService/Controllers/TaxInfoController.cs [hunk 4]
```diff
@@ -1,52 +1,44 @@
...
         private readonly ILogger<TaxInfoController> _logger;
+        private readonly TaxInfoProviderConfiguration _cuitServiceConfiguration;
+        private readonly IFlurlClient _flurClient;
 
-        public TaxInfoController(ILogger<TaxInfoController> logger)
+        public TaxInfoController(ILogger<TaxInfoController> logger, IOptions<TaxInfoProviderConfiguration> cuitServiceConfiguration)
         {
             _logger = logger;
+            _cuitServiceConfiguration = cuitServiceConfiguration.Value;
+            _flurClient = new FlurlClient().AllowAnyHttpStatus();
```
Comment (nreal) [inline CuitService/Controllers/TaxInfoController.cs:22]: What do you think to inject IFluentClient in the constructor?

Comment (andresmoschini) [reply CuitService/Controllers/TaxInfoController.cs:22]: I agree with @nreal that we should eventually improve this code, maybe removing _Flurl_ completely in favor of _HttpClient_.

But by the moment I think that current implementation is acceptable, or we could even simplify it a little more using [Fluent HTTP](https://flurl.dev/docs/fluent-http/) (without using `new`).

Internally Flurl has smart management of _HttpClient_ and thanks to _HttpTest_, we do not need to inject _IFlurlClient_ to test it.

Comment (andresmoschini) [reply CuitService/Controllers/TaxInfoController.cs:22]: Why do you need to accept any HTTP status?

Comment (andresmoschini) [reply CuitService/Controllers/TaxInfoController.cs:22]: Here you have an example without explicit _FlurlClient_ instances:

image

In my opinion, it is not so much flexible, but we do not need such flexibility (at least not yet)

File: CuitService.Test/WebApplicationFactoryHelper.cs [hunk 5]
```diff
@@ -22,5 +22,12 @@ public static WebApplicationFactory<Startup> ConfigureSecurityOptions(this WebAp
         public static WebApplicationFactory<Startup> WithDisabledLifeTimeValidation(this WebApplicationFactory<Startup> factory)
             => factory.ConfigureSecurityOptions(
                 o => o.SkipLifetimeValidation = true);
+
+        public static WebApplicationFactory<Startup> WithPreservedExecutionContext(this WebApplicationFactory<Startup> factory)
+        {
+            factory.Server.PreserveExecutionContext = true;
+
+            return factory;
+        }
```
Comment (andresmoschini) [inline CuitService.Test/WebApplicationFactoryHelper.cs:31]: Creo que no fue buena idea incluir este helper, porque no puede ir en cualquier parte de la construcción del client, sino justo al final. Por otro lado, necesitamos hacer _dispose_ del factory, así que yo quitaría este helper.

Más detalles en un comentario en un test.

**Comment by andresmoschini** [comment]
👍

**Pull request merged**